huaweicloud/services/workspace/data_source_huaweicloud_workspace_desktop_tags_filter.go

```diff
@@ -0,0 +1,284 @@
+			"key": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The key of tag.",
+			},
+			"value": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The value of tag.",
+			},
+		},
+	}
+}
+
+func desktopQueriedMatchFilterSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"key": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of desktop property.",
+			},
+			"value": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The value of desktop property.",
+			},
+		},
+	}
+}
+
+func buildDesktopTagsFilter(d *schema.ResourceData) (map[string]interface{}, error) {
+	filter := map[string]interface{}{
+		"action": "filter",
+		"offset": "0",
+		"limit":  "200",
+	}
+
+	if v, ok := d.GetOk("without_any_tag"); ok {
+		filter["without_any_tag"] = v.(bool)
+	}
+
+	if v, ok := d.GetOk("matches"); ok {
+		matches := make([]map[string]interface{}, 0)
+		for _, item := range v.([]interface{}) {
+			match := item.(map[string]interface{})
+			matches = append(matches, map[string]interface{}{
+				"key":   match["key"].(string),
+				"value": match["value"].(string),
+			})
+		}
+		filter["matches"] = matches
+	}
+
+	if v, ok := d.GetOk("tags"); ok {
+		filter["tags"] = buildTagsCondition(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("tags_any"); ok {
+		filter["tags_any"] = buildTagsCondition(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("not_tags"); ok {
+		filter["not_tags"] = buildTagsCondition(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("not_tags_any"); ok {
+		filter["not_tags_any"] = buildTagsCondition(v.([]interface{}))
+	}
+
+	return filter, nil
+}
+
+func buildTagsCondition(tagsList []interface{}) []map[string]interface{} {
+	tags := make([]map[string]interface{}, 0)
+	for _, item := range tagsList {
+		tag := item.(map[string]interface{})
+		values := make([]string, 0)
+		for _, v := range tag["values"].([]interface{}) {
+			values = append(values, v.(string))
+		}
+		tags = append(tags, map[string]interface{}{
+			"key":    tag["key"].(string),
+			"values": values,
+		})
+	}
+	return tags
+}
+
+func flattenDesktops(resources []interface{}) []interface{} {
+	if len(resources) < 1 {
+		return nil
+	}
+
+	result := make([]interface{}, 0, len(resources))
+	for _, item := range resources {
+		resource := item.(map[string]interface{})
+		desktop := map[string]interface{}{
+			"resource_id":     utils.PathSearch("resource_id", resource, nil),
+			"resource_name":   utils.PathSearch("resource_name", resource, nil),
+			"resource_detail": utils.PathSearch("resource_detail", resource, nil),
+			"tags":            flattenDesktopTags(utils.PathSearch("tags", resource, nil).([]interface{})),
+		}
+		result = append(result, desktop)
+	}
+	return result
+}
+
+func filterDesktopsByTags(client *golangsdk.ServiceClient, filter map[string]interface{}) ([]interface{}, error) {
```

Comment:
define offset and limit parameters in this function and using for condition to query tag list.